### PR TITLE
Oslo config

### PR DESCRIPTION
**Pull request by dstepanenko**

### CODE REVIEW

File: kostyor/db/api.py [hunk 1]
```diff
@@ -11,18 +12,35 @@
 
 
 # TODO (sc68cal) save the database file in a configurable location
```
Comment (jvarlamova) [inline kostyor/db/api.py]: Seems that now we can remove this comment.

File: kostyor/conf.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from oslo_config import cfg
+config_files = ['conf.ini']
```
Comment (ikalnitsky) [inline kostyor/conf.py]: Hm.. I think we don't need the `conf.ini` if we're going to use `oslo.cfg`. How about follow OpenStack approach and set defaults in python code within `default_opts` below?

File: kostyor/rest_api.py [hunk 3]
```diff
@@ -308,4 +309,7 @@ def cluster_list():
 
 
 if __name__ == '__main__':
+    conf.parse_args()
+    engine = create_engine(conf.CONF.database, convert_unicode=True)
```
Comment (ikalnitsky) [inline kostyor/rest_api.py]: As you moved `db_session.remove()` out of `rest_api.py` can you also moved out `create_engine`? I think something like `db_api.configure_session(conf.CONF.database)` should work.

File: conf.ini [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[DEFAULT]
+database = sqlite:////tmp/kostyor.db
```
Comment (ikalnitsky) [inline conf.ini]: Database often include bunch of other options to configure such as connection pool, for example. So I'd prefer to have the `database` section with `connection` option in order to easily extend it if needed.

```
[database]
conection = sqlite:///tmp/kostyor.db
```

Does it make sense?

File: kostyor/conf.py (hunk 2 above)
Comment (sc68cal) [inline kostyor/conf.py]: Agree, let's put the defaults in this file and then have `rest_api.py` have an optional argument to the path. Flask already has an environment variable that it uses for configuration. We may be able to utilize.

http://flask.pocoo.org/docs/0.11/config/

File: kostyor/db/api.py [hunk 5]
```diff
@@ -23,6 +20,16 @@ def _get_most_recent_upgrade_task(cluster_id):
     return q.first()
 
 
+def configure_session(database):
+    engine = create_engine(database, convert_unicode=True)
+    db_session.configure(bind=engine)
```
Comment (sc68cal) [inline kostyor/db/api.py]: Nice find, that solves the issue of having this module imported

File: kostyor/rest_api.py [hunk 6]
```diff
@@ -308,4 +309,6 @@ def cluster_list():
 
 
 if __name__ == '__main__':
+    conf.parse_args(sys.argv[1:])
+    db_api.configure_session(conf.CONF.database.connection)
```
Comment (ikalnitsky) [inline kostyor/rest_api.py]: minor: think it's better to pass the whole database config (`conf.CONF.database`).

File: kostyor/conf.ini [hunk 7]
```diff
@@ -1,2 +1,2 @@
-[global]
-discovery_methods = first,second
+[database]
```
Comment (ikalnitsky) [inline kostyor/conf.ini]: do we use this file somewhere? or it's kept as an example?

**Comment by sc68cal** [comment]
@dstepanenko if you can rebase this and fix the merge conflict, I'll merge this asap

**Comment by sc68cal** [comment]
@dstepanenko  	317a738 appears to be a merge commit. You need to do a `rebase`

**Pull request merged**